src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -134,7 +137,25 @@ export class FocusMonitor implements OnDestroy {
     this._windowFocusTimeoutId = setTimeout(() => this._windowFocused = false);
   }
 
-  constructor(private _ngZone: NgZone, private _platform: Platform) {}
+   /** Used to reference correct document/window */
+   protected _document?: Document;
+
+  constructor(private _ngZone: NgZone,
+              private _platform: Platform,
+              /** @breaking-change 11.0.0 make document required */
+              @Optional() @Inject(DOCUMENT) document?: any) {
+    this._document = document;
+  }
+
+  /** Access injected document if available or fallback to global document reference */
+  get document(): Document {
```

Comment:
This should be a private API (here and elsewhere)

Reply:
Ok sounds good. I'll refactor to private functions _getDocument() and _getWindow()